atomate/vasp/firetasks/exchange_tasks.py

```diff
@@ -0,0 +1,458 @@
+
+                # J_ij exists
+                if len(ex_params) > 1:
+                    E0 = abs(ex_params["E0"])
+                    for k, v in ex_params.items():
+                        if k != "E0":
+                            if abs(v) > E0:  # |J_ij| > |E0| unphysical!
+                                converged = False
+                else:  # Only <J> was computed
+                    converged = False
+
+                if converged:
+                    converged_list.append(hmodel.as_dict())
+
+            # If multiple Heisenberg Models converged, take the maximal cutoff
+            if len(converged_list) > 0:
+                hmodel = converged_list[-1]  # Largest cutoff
+            else:  # No converged models
+                hmodel = None
+
+        # Update FW spec with converged hmodel or None
+        update_spec = {"converged_heisenberg_model": hmodel}
+
+        return FWAction(update_spec=update_spec)
+
+
+@explicit_serialize
+class VampireMC(FiretaskBase):
+    """
+    If there is a suitable Heisenberg model, run Monte Carlo to compute the critical temperature.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        exchange_wf_uuid (int): Unique id for record keeping.
+        parent_structure (Structure): Structure object with magmoms.
+        mc_settings (dict): A configuration dict for monte carlo.
+        avg (bool): Only <J> exchange param.
+
+    TODO:
+        * Include HeisenbergModel convergence check.
+
+    """
+
+    required_params = [
+        "db_file",
+        "exchange_wf_uuid",
+        "parent_structure",
+        "mc_settings",
+        "avg",
+    ]
+    optional_params = []
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["exchange_wf_uuid"]
+        formula_pretty = self["parent_structure"].composition.reduced_formula
+        mc_settings = self["mc_settings"]
+        avg = self["avg"]
+
+        # Get Heisenberg models from db
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+        mmdb.collection = mmdb.db["exchange"]
+
+        # Get documents
+        docs = list(
+            mmdb.collection.find(
+                {"wf_meta.wf_uuid": wf_uuid}, ["heisenberg_model", "nn_cutoff"]
+            )
+        )
+
+        hmodels = [HeisenbergModel.from_dict(d["heisenberg_model"]) for d in docs]
+        hmodel = hmodels[0]  # Take the model with smallest NN cutoff
+
+        task_doc = {"wf_meta": {"wf_uuid": wf_uuid}, "formula_pretty": formula_pretty}
+
+        if fw_spec.get("tags", None):
+            task_doc["tags"] = fw_spec["tags"]
+
+        # Vampire monte carlo settings
+        mc_box_size = mc_settings["mc_box_size"]
+        equil_timesteps = mc_settings["equil_timesteps"]
+        mc_timesteps = mc_settings["mc_timesteps"]
+
+        # Get a converged Heisenberg model if one was found
+        # if fw_spec["converged_heisenberg_model"]:
```

Comment:
Is there a reason why you don't use this approach?

Reply:
At the moment, because of the issues with the input static calculations, this sort of convergence testing is really not very informative or useful. I think with constrained magnetic moments it will be worth doing.